### ISSUE TITLE
Tests are loading all the backends

### ISSUE THREAD

**Issue by gbin** [gbin/err#434]
RobSpectre: get the sense I'm not specifying the backend in the tests correctly so it is loading all of them.
Looks like in the errbot tests you copy the config to the local directory - doesn't seem like these tests are picking that up.
Here's my tox config: https://github.com/RobSpectre/err-twilio-lookup/blob/master/tox.ini
Not sure what I'm missing here.

**Comment by RobSpectre** [comment]
Central issue appears to be the path PluginFileLocator is using to look for plugins. When I set my `extra_plugin_dir` to `'.'` it appears to search my entire path:

```
INFO:Backends search paths ['/home/rspectre/.virtualenvs/err/lib/python3.4/site-packages/errbot/backends']
```

**Comment by RobSpectre** [comment]
Full output of where PluginFileLocator is looking:

```
(err)rspectre@rockettorussia:~/workspace/err-twilio-lookup$ py.test -x | grep "PluginFileLocator walks (recursively)"
DEBUG:PluginFileLocator walks (recursively) into directory: /home/rspectre/.virtualenvs/err/lib/python3.4/site-packages/errbot/backends
DEBUG:PluginFileLocator walks (recursively) into directory: /home/rspectre/.virtualenvs/err/lib/python3.4/site-packages/errbot/backends
DEBUG:PluginFileLocator walks (recursively) into directory: /home/rspectre/.virtualenvs/err/lib/python3.4/site-packages/errbot/core_plugins
DEBUG:PluginFileLocator walks (recursively) into directory: /home/rspectre/workspace/err-twilio-lookup/.tox/py34/lib/python3.4/site-packages/errbot/backends
DEBUG:PluginFileLocator walks (recursively) into directory: /home/rspectre/workspace/err-twilio-lookup
DEBUG:PluginFileLocator walks (recursively) into directory: /home/rspectre/workspace/err-twilio-lookup/.tox/py34/lib/python3.4/site-packages/errbot/core_plugins
```

**Comment by RobSpectre** [comment]
Root cause here is that the test runner I was using `tox` created the virtualenvs for its tests in the same directory as the plugin project, caused `PluginFileLocator` to run recursively in each of them and find each backend in the `errbot`` directory.

```
DEBUG:PluginFileLocator walks (recursively) into directory: /home/rspectre/workspace/err-twilio-lookup/.tox/py34/lib/python3.4/site-packages/errbot/backends
DEBUG:PluginFileLocator walks (recursively) into directory: /home/rspectre/workspace/err-twilio-lookup
DEBUG:PluginFileLocator walks (recursively) into directory: /home/rspectre/workspace/err-twilio-lookup/.tox/py34/lib/python3.4/site-packages/errbot/core_plugins
```
Solution is to add the following to the `[tox]` section of your `tox.ini`:

`toxworkdir = /tmp/err-twilio-lookup/.tox`

Can close this issue.

**Comment by gbin** [comment]
This was properly fixed by 51122890226ddb9eea7b494fb1e344d87753fe1b

Now we stop the recursion at dotted directories, it makes a lot of sense not only for tor but for example with .git
